cosmovisor/upgrade.go

```diff
@@ -69,16 +69,64 @@ func DownloadBinary(cfg *Config, info *UpgradeInfo) error {
 		// copy binary to binPath from dirPath if zipped directory don't contain bin directory to wrap the binary
 		if err != nil {
 			err = copy.Copy(filepath.Join(dirPath, cfg.Name), binPath)
+		}
+		if err != nil {
+			// If there is only one entry in dirPath, then promote it to the upgradedir
+
+			// Clean up an empty bin directory.
+			os.Remove(filepath.Join(dirPath, "bin"))
```

Comment:
Explicitly ignore the return value with `_ := os.Remove(...)` so we can someday turn on warnings for unhandled return values.